internal/resources/google/compute_cost_component_helpers.go

```diff
@@ -33,19 +33,21 @@ type ContainerNodeConfig struct {
 // computeCostComponent returns a cost component for Compute instance usage.
 func computeCostComponents(region, machineType string, purchaseOption string, instanceCount int64, monthlyHours *float64) ([]*schema.CostComponent, error) {
 	if strings.HasPrefix(strings.ToLower(machineType), "e2-custom") {
-		return nil, errors.New("Infracost currently does not support E2 custom instances")
+		return nil, errors.New("infracost currently does not support E2 custom instances")
 	}
 
 	sustainedUseDiscount := 0.0
 	fixPurchaseOption := ""
 
-	if strings.ToLower(purchaseOption) == "on_demand" {
+	if strings.ToLower(purchaseOption) == "on_demand" && monthlyHours != nil {
 		fixPurchaseOption = "OnDemand"
 		switch strings.ToLower(strings.Split(machineType, "-")[0]) {
 		case "c2", "n2", "n2d":
-			sustainedUseDiscount = 0.2
+			sustainedUseDiscount = getSustainedUseDiscount(monthlyHours, []float64{0, 0.25, 0.50, 0.75}, []float64{1.0, 0.8678, 0.733, 0.6}, []float64{0.2088, 0.1811, 0.1530, 0.1252})
 		case "custom", "n1", "f1", "g1", "m1":
-			sustainedUseDiscount = 0.3
+			sustainedUseDiscount = getSustainedUseDiscount(monthlyHours, []float64{0, 0.25, 0.50, 0.75}, []float64{1.0, 0.80, 0.60, 0.40}, []float64{0.0475, 0.0380, 0.0285, 0.0190})
+		default:
+			return nil, errors.New("unsupported machine type for sustained use discount")
```

Comment:
I think if we don't support sustained use discounts for a machine type we should fall back to a 0% discount.